src/commands/apphosting-secrets-grantaccess.ts

```diff
@@ -0,0 +1,113 @@
+import { Command } from "../command";
+import { Options } from "../options";
+import { needProjectId, needProjectNumber } from "../projectUtils";
+import { FirebaseError } from "../error";
+import { requireAuth } from "../requireAuth";
+import * as secrets from "../functions/secrets";
+import { requirePermissions } from "../requirePermissions";
+import { logger } from "../logger";
+import * as apphosting from "../gcp/apphosting";
+import * as secretManager from "../gcp/secretManager";
+import { setIamPolicy } from "../gcp/secretManager";
+import * as iam from "../gcp/iam";
+
+export const command = new Command("apphosting:secrets:grantaccess <secretName>")
+  .description("grant service accounts permissions to the provided secret")
+  .option("-l, --location <location>", "app backend location")
+  .option("-b, --backend <backend>", "app backend name")
+  .before(requireAuth)
+  .before(secrets.ensureApi)
+  .before(apphosting.ensureApiEnabled)
+  .before(requirePermissions, [
+    "secretmanager.secrets.create",
+    "secretmanager.secrets.get",
+    "secretmanager.secrets.update",
+    "secretmanager.versions.add",
+    "secretmanager.secrets.setIamPolicy",
+  ])
+  .action(async (secretName: string, options: Options) => {
+    const projectId = needProjectId(options);
+    const projectNumber = await needProjectNumber(options);
+
+    if (!options.location) {
+      logger.error(
```

Comment:
whats the difference between logger.error() and throwing a FirebaseError?

Reply:
Looks like logger.error() is just more poorly formatted for our use case, changed to FirebaseError